obspy/clients/fdsn/mass_downloader/download_helpers.py

```diff
@@ -696,7 +706,17 @@ def star_download_station(args):
                 starttime = min([_i.starttime for _i in c_info])
                 endtime = max([_i.endtime for _i in c_info])
                 if starttime > times[0] or endtime < times[1]:
-                    still_missing[c_id] = times
+                    # Cope with case that not full day of station info missing
+                    if starttime < times[1]:
+                        still_missing[c_id] = (times[0], starttime)
+                        station.have_station_information[c_id] = (starttime,
+                                                                  times[1])
+                    elif endtime > times[0]:
+                        still_missing[c_id] = (endtime, times[1])
+                        station.have_station_information[c_id] = (times[0],
+                                                                  endtime)
+                    else:
+                        still_missing[c_id] = times
```

Comment:
This is actually already along the lines I proposed above before I fully read your PR - sorry about that.

It currently does not deal with the case if information at the beginning and end is missing. I think this might be easiest to implement with a new attribute as I proposed above but maybe this is another way.